### ISSUE TITLE
Add support for Terraform v1.0

### ISSUE THREAD

**Issue by PaulinaMoreno** [terraform-google-modules/terraform-google-cloud-datastore#28]
### TL;DR

Hi, I'm trying to deploy the **simple_example** and when I run "terraform init" I have the next error:

`Error: Invalid reference from destroy provisioner
on ../../main.tf line 37, in resource "null_resource" "cloud-datastore-indices":
37:     command = "${path.module}/scripts/destroy-indexes.sh '${var.project}' '${local.null_index_path_file}'"

Destroy-time provisioners and their connection configurations may only reference attributes of the related
resource, via 'self', 'count.index', or 'each.key'.

References to other resources during the destroy phase can cause dependency cycles and interact poorly with
create_before_destroy.`

### Terraform Configuration

```hcl
None
```

### Terraform Version

```sh
Terraform v1.0.6
on darwin_amd64
```

### Additional information

I suppose that the error is related to the version of terraform and how data in reference in Terraform 0.13+.

**Comment by apeabody** [comment]
Hi @PaulinaMoreno Thanks for reaching out!  It appears there are several updates that need to be addressed to support the newer Terraform versions.  I hope to look at them in the next few weeks:

#25 
#26
